### PR TITLE
init: initialize the project

### PR DESCRIPTION
**Pull request by Jeffzholy**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module.exports = {
+    "env": {
+        "browser": true,
+        "es6": true,
+        "node": true,
+        "jest":true
+    },
+    "extends": [
+        "plugin:react/recommended",
```
Comment (moonavw) [inline .eslintrc.js]: just extends airbnb

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        "airbnb"
+    ],
+    "globals": {
+        "Atomics": "readonly",
+        "SharedArrayBuffer": "readonly"
+    },
+    "parserOptions": {
```
Comment (moonavw) [inline .eslintrc.js]: `parser: "babel-eslint"`,  and these parserOptions already defined by them, e.g. jsx by airbnb, ecma by babel's, etc

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        "ecmaFeatures": {
+            "jsx": true
+        },
+        "ecmaVersion": 2018,
+        "sourceType": "module"
+    },
+    "plugins": [
+        "react"
```
Comment (moonavw) [inline .eslintrc.js]: airbnb already take it

File: .gitignore [hunk 4]
```diff
@@ -1,104 +1,3 @@
-# Logs
-logs
-*.log
-npm-debug.log*
-yarn-debug.log*
```
Comment (moonavw) [inline .gitignore]: the auto generated gitignore is good enough to go, no need to change

**Comment by moonavw** [comment]
missing the html file, try this https://webpack.js.org/plugins/html-webpack-plugin/

File: frontend/components/App.test.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { shallow, configure } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+import App from './App';
+
+configure({ adapter: new Adapter() });
```
Comment (moonavw) [inline frontend/components/App.test.js]: move to separate jest setup file, apply to jest config

File: .eslintrc.js (hunk 1 above)
Comment (Jeffzholy) [reply .eslintrc.js]: @moonavw "plugin:react/recommended" comes from eslint --init

File: .eslintrc.js (hunk 2 above)
Comment (Jeffzholy) [reply .eslintrc.js]: @moonavw still comes from eslint --init

File: .eslintrc.js (hunk 3 above)
Comment (Jeffzholy) [reply .eslintrc.js]: @moonavw  still comes from eslint --init

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    steps:
+    - uses: actions/checkout@v1
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm i yarn -g
+    - run: yarn install
+    - run: yarn run lint
+    - run: yarn run test
```
Comment (moonavw) [inline .github/workflows/ci.yml]: 1. yarn lint
2. yarn test
3. yarn build

File: package.json [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  "version": "1.0.0",
+  "main": "index.js",
+  "repository": "git@github.com:Jeffzholy/Mentoring.git",
+  "license": "MIT",
+  "scripts": {
+    "lint": "eslint '**/*.js'",
+    "lint-fix": "eslint --fix '**/*.js'",
+    "test": "jest",
+    "test-watch": "jest --watch",
+    "build-backend": "babel backend/app.js -d dist/backend",
```
Comment (moonavw) [inline package.json]: same prefixed scripts should be placed closely. also need `build`
```
build: yarn build-backend && yarn build-frontend
build-backend:
build-frontend:
```

File: webpack.config.common.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+
+module.exports = {
+  entry: './frontend/index.js',
+  output: {
+    filename: 'bundle.js',
```
Comment (moonavw) [inline webpack.config.common.js]:
```suggestion
    filename: '[name].js',
```

File: webpack.config.prod.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+const merge = require('webpack-merge');
+const common = require('./webpack.config.common.js');
+
+module.exports = merge(common, {
+  mode: 'production',
+  output: { filename: '[name].[hash].js' },
```
Comment (moonavw) [inline webpack.config.prod.js]:
```suggestion
  output: { filename: '[name].[contenthash].bundle.js' },
```

**Pull request merged**